src/util/state-management.js

```diff
@@ -69,7 +81,8 @@ export function getStatefulPropsContext(reducers, { getState, setState }) {
 export function buildStatefulComponent(baseComponent, opts) {
 	opts = _.merge({
 		setStateWithNewProps: true, // if true, new props will update state, else prop has priority over existing state
-		replaceEvents: false // if true, function props replace the existing reducers, else they are invoked *after* state reducer returns
+		replaceEvents: false, // if true, function props replace the existing reducers, else they are invoked *after* state reducer returns
+		stateBeatsProps: false, // if true, when the component re-renders, state will take precedence over props, but on componentwillreceiveprops it will still favor props, there is a very limited case where this useful, mostly with text inputs to keep the user input until new props come in
```

Comment:
This is the new option. It's got a useful comment:

> when the component re-renders, state will take precedence over props, but on componentwillreceiveprops it will still favor props, there is a very limited case where this useful, mostly with text inputs to keep the user input until new props come in